internal/services/storage/client/data_plane.go

```diff
@@ -44,9 +44,15 @@ func (Client) DataPlaneOperationSupportingOnlySharedKeyAuth() DataPlaneOperation
 	}
 }
 
-func (c Client) configureDataPlane(ctx context.Context, clientName string, baseClient client.BaseClient, account accountDetails, operation DataPlaneOperation) error {
-	if operation.SupportsAadAuthentication && c.authorizerForAad != nil {
-		baseClient.SetAuthorizer(c.authorizerForAad)
+func (c Client) configureDataPlane(ctx context.Context, clientName, resourceIdentifier string, baseClient client.BaseClient, account accountDetails, operation DataPlaneOperation) error {
+	if operation.SupportsAadAuthentication && c.authConfig != nil {
+		api := c.authConfig.Environment.Storage.WithResourceIdentifier(resourceIdentifier)
```

Comment:
I suspect this might be problematic since we're mutating the original `Storage` object - see https://github.com/hashicorp/go-azure-sdk/pull/960#discussion_r1560618894